host_applications/linux/apps/raspicam/RaspiVid.c

```diff
@@ -806,6 +844,40 @@ static int parse_cmdline(int argc, const char **argv, RASPIVID_STATE *state)
          break;
       }
 
+      case CommandRaw:  // output filename
+      {
+         state->raw_output = 1;
+         state->raw_output_fmt = RAW_OUTPUT_FMT_YUV;
+         int len = strlen(argv[i + 1]);
+         if (len)
+         {
+            state->raw_filename = malloc(len + 1);
+            vcos_assert(state->raw_filename);
+            if (state->raw_filename)
+               strncpy(state->raw_filename, argv[i + 1], len+1);
+            i++;
+         }
+         else
+            valid = 0;
+         break;
+      }
+
+      case CommandRawFormat:
+      {
+         state->raw_output_fmt = raspicli_map_xref(argv[i + 1], raw_output_fmt_map, raw_output_fmt_map_size);
+         i++;
+
+         if (state->raw_output_fmt == -1)
+            state->raw_output_fmt = RAW_OUTPUT_FMT_YUV;
+
+         if (!state->raw_output)
+         {
+            fprintf(stderr, "--raw-fmt is only valid with --raw output selected\n");
+            valid = 0;
```

Comment:
Ah the old chicken and egg problem.
```raspivid --raw filename.rgb --raw-format rgb``` is valid, but
```raspivid --raw-format rgb --raw filename.rgb ``` isn't because output wasn't set when they tried setting the format. There's never a good way around that, except possibly checking the two settings after they've all been parsed and then identifying the conflict.

And line 850 would mean that you'd get YUV with the second line anyway.